src/core_plugins/metrics/public/components/vis_types/timeseries/vis.js

```diff
@@ -89,6 +91,7 @@ function TimeseriesVisualization(props) {
         axisCount++;
 
         const formatter = tickFormatter(row.formatter, row.value_template);
+        formatter.key = row.formatter + row.value_template;
```

Comment:
This is some groovy JS stuff :stuck_out_tongue: !

- this primary-key function produces odd values that just work because JS isn't picky 
    - value_template is undefined by default, so you get things like `bytesundefined` and `numberundefined`. Similarly, you get the same with an un-templated custom-formatter, things like `null{{value}}` .
    - `bytes{{value}}` and `bytesundefined` should really key to the same.
- We are assigning a string property to an anonymous function. And then we read that out in a different file. it's a little obscure.

We already have a y-axis object that we pass around and has some properties that are used to make the determination of when to change. So why not just assign these properties to the `yaxis`-object and do the equality-check in `shouldComponentUpdate` in flot_chart?

e.g.
```
yaxis = {
      alignTicksWithAxis: 1,
      position: row.axis_position,
       tickFormatter: formatter,
       axis_formatter: row.axis_formatter,
      formatterName: row.formatter,
     formatterValueTemplate: row.value_template
}
```

and then evaluate those here instead: https://github.com/thomasneirynck/kibana/blob/b2ab7ca9480b68971339dc78ae61c961cf2a3696/src/core_plugins/metrics/public/visualizations/components/flot_chart.js#L29